pyannote/database/loader.py

```diff
@@ -187,6 +187,55 @@ def __call__(self, file: ProtocolFile) -> Timeline:
         return self.loaded_[uri]
 
 
+class LABLoader:
+    """LAB loader
+
+    Parameters
+    ----------
+    path : str
+        Path to LAB file with mandatory {uri} placeholder.
+        (e.g. "/path/to/{uri}.lab")
+
+        each .lab file contains the segments for a single audio file, in the following format:
+        start end label
+
+        ex.
+        0.0 12.3456 sing
+        12.3456 15.0 nosing
+        ...
+    """
+
+    def __init__(self, path: Text = None):
+        super().__init__()
+
+        self.path = str(path)
+        
+        _, placeholders, _, _ = zip(*string.Formatter().parse(self.path))
+        self.placeholders_ = set(placeholders) - set([None])
+        self.loaded_ = dict()
+        
+    def __call__(self, file: ProtocolFile) -> Annotation:
+
+        uri = file["uri"]
+
+        if uri in self.loaded_:
+            return self.loaded_[uri]
+
+        sub_file = {key: file[key] for key in self.placeholders_}
+        loaded = load_lab(self.path.format(**sub_file), uri)
+        
+        # do not cache timelines when there is one LAB file per "uri"
+        # since loading it should be quite fast
+        if "uri" in self.placeholders_:
```

Comment:
This will always be true because there is no other way to obtain the `uri`.
We should therefore return `loaded` automatically. The `if` statement can be removed.